common_tools/get_dumps.py

```diff
@@ -0,0 +1,40 @@
+import os
+from pathlib import Path
+import argparse
+
+
+def compile_dumps(mlir_dir: str):
+    dumps_dir = Path(args.dir).parent.joinpath("dumps")
+    i_o_paths = []
+    for root, _, files in os.walk(mlir_dir):
```

Comment:
You can use Path.walk so that we don't keep mixing Path and os: https://docs.python.org/3/library/pathlib.html#pathlib.Path.walk